src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -59,59 +59,8 @@ FixedwingPositionControl::FixedwingPositionControl(bool vtol) :
-	_parameter_handles.roll_limit = param_find("FW_R_LIM");
-	_parameter_handles.throttle_min = param_find("FW_THR_MIN");
-	_parameter_handles.throttle_max = param_find("FW_THR_MAX");
-	_parameter_handles.throttle_idle = param_find("FW_THR_IDLE");
-	_parameter_handles.throttle_slew_max = param_find("FW_THR_SLEW_MAX");
-	_parameter_handles.throttle_cruise = param_find("FW_THR_CRUISE");
-	_parameter_handles.throttle_alt_scale = param_find("FW_THR_ALT_SCL");
-	_parameter_handles.throttle_land_max = param_find("FW_THR_LND_MAX");
-	_parameter_handles.man_roll_max_deg = param_find("FW_MAN_R_MAX");
-	_parameter_handles.man_pitch_max_deg = param_find("FW_MAN_P_MAX");
-	_parameter_handles.rollsp_offset_deg = param_find("FW_RSP_OFF");
-	_parameter_handles.pitchsp_offset_deg = param_find("FW_PSP_OFF");
-
-	_parameter_handles.land_slope_angle = param_find("FW_LND_ANG");
-	_parameter_handles.land_H1_virt = param_find("FW_LND_HVIRT");
-	_parameter_handles.land_flare_alt_relative = param_find("FW_LND_FLALT");
-	_parameter_handles.land_flare_pitch_min_deg = param_find("FW_LND_FL_PMIN");
-	_parameter_handles.land_flare_pitch_max_deg = param_find("FW_LND_FL_PMAX");
-	_parameter_handles.land_thrust_lim_alt_relative = param_find("FW_LND_TLALT");
-	_parameter_handles.land_heading_hold_horizontal_distance = param_find("FW_LND_HHDIST");
-	_parameter_handles.land_use_terrain_estimate = param_find("FW_LND_USETER");
-	_parameter_handles.land_early_config_change = param_find("FW_LND_EARLYCFG");
-	_parameter_handles.land_airspeed_scale = param_find("FW_LND_AIRSPD_SC");
-	_parameter_handles.land_throtTC_scale = param_find("FW_LND_THRTC_SC");
-
-	_parameter_handles.time_const = param_find("FW_T_TIME_CONST");
-	_parameter_handles.time_const_throt = param_find("FW_T_THRO_CONST");
-	_parameter_handles.min_sink_rate = param_find("FW_T_SINK_MIN");
-	_parameter_handles.max_sink_rate = param_find("FW_T_SINK_MAX");
-	_parameter_handles.max_climb_rate = param_find("FW_T_CLMB_MAX");
-	_parameter_handles.climbout_diff = param_find("FW_CLMBOUT_DIFF");
-	_parameter_handles.throttle_damp = param_find("FW_T_THR_DAMP");
-	_parameter_handles.integrator_gain = param_find("FW_T_INTEG_GAIN");
-	_parameter_handles.vertical_accel_limit = param_find("FW_T_VERT_ACC");
-	_parameter_handles.speed_comp_filter_omega = param_find("FW_T_SPD_OMEGA");
-	_parameter_handles.roll_throttle_compensation = param_find("FW_T_RLL2THR");
-	_parameter_handles.speed_weight = param_find("FW_T_SPDWEIGHT");
-	_parameter_handles.pitch_damping = param_find("FW_T_PTCH_DAMP");
-	_parameter_handles.heightrate_p = param_find("FW_T_HRATE_P");
-	_parameter_handles.heightrate_ff = param_find("FW_T_HRATE_FF");
-	_parameter_handles.speedrate_p = param_find("FW_T_SRATE_P");
-	_parameter_handles.loiter_radius = param_find("NAV_LOITER_RAD");
+	// if vehicle is vtol these handles will be set when we get the vehicle status
+	_parameter_handles.airspeed_trans = PARAM_INVALID;
```

Comment:
Why is setting it here to PARAM_INVALID necessary? Isn't it already set correctly above?

Side note: is there actually something preventing us from using the new params API also in the VTOL module? If not then I'm happy helping out with that.

Reply:
It's not, this looks bogus.

> Side note: is there actually something preventing us from using the new params API also in the VTOL module? If not then I'm happy helping out with that.

Just time. :smirk: